walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,75 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
+  /**
+   * Get the collection of time ranges when an event can be held.
+   * @param events
+   * @param request
+   * @return Returns the collection of time ranges when the event can be held.
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+
+    Collection<TimeRange> validTimeRanges = new ArrayList<TimeRange>();
+
+    Collection<TimeRange> orderedConflicts = 
+        getOrderedAttendingEvents(events, request.getAttendees());
+    Iterator<TimeRange> conflictsIterator = orderedConflicts.iterator();
+    int prevConflictEndTime = TimeRange.START_OF_DAY;
+
+    while (conflictsIterator.hasNext()) {
+      TimeRange conflict = conflictsIterator.next();
+      int conflictStartTime = conflict.start();
+      
+      // Check for gap between prev conflict and this conflict.
+      if (conflictStartTime > prevConflictEndTime && 
```

Comment:
Is the first check necessary? If conflictStartTime is less than prevConflictEndTime wouldn't it be caught by the second check anyways?

Reply:
Funny you bring that up. That was my initial thought until I saw your code. I'm imagining that there could be an error here given a negative duration in the event request, although now that I think about it not only is that probably not going to happen but also there'd be a lot more to worry about if that were the case.